File: backend/app/schemas/analytics.py
from pydantic import BaseModel

class TrendData(BaseModel):
    date: str
    count: int

class CategoryDistribution(BaseModel):
    category: str
    count: int

class SeverityDistribution(BaseModel):
    severity: str
    count: int

class AnalyticsResponse(BaseModel):
    total_reviews: int
    reviews_this_week: int
    reviews_this_month: int
    trends: list[TrendData]
    category_distribution: list[CategoryDistribution]
    severity_distribution: list[SeverityDistribution]

File: frontend/src/types/analytics.ts
export interface TrendData {
  date: string;
  count: number;
}

export interface CategoryDistribution {
  category: string;
  count: number;
}

export interface SeverityDistribution {
  severity: string;
  count: number;
}

export interface AnalyticsResponse {
  total_reviews: number;
  reviews_this_week: number;
  reviews_this_month: number;
  trends: TrendData[];
  category_distribution: CategoryDistribution[];
  severity_distribution: SeverityDistribution[];
}

export interface ReviewListItem {
  id: number;
  review_type: string;
  source_name: string;
  quality_score: number;
  total_issues: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  model_used: string;
  duration_seconds: number;
  created_at: string;
}

export interface ReviewListResponse {
  reviews: ReviewListItem[];
  total: number;
  page: number;
  limit: number;
}
